sipuadart/lib/sdp/attributes/attribute.dart

```diff
@@ -0,0 +1,67 @@
+
+import 'package:sipuadart/sdp/field.dart';
+
+/// https://tools.ietf.org/html/rfc4566#page-24
+/// # Attributes
+///
+/// Attributes are the primary means for extending SDP.  Attributes may
+/// be defined to be used as "session-level" attributes, "media-level"
+/// attributes, or both.
+///
+/// A media description may have any number of attributes ("a=" fields)
+/// which are media specific.  These are referred to as "media-level"
+/// attributes and add information about the media stream.  Attribute
+/// fields can also be added before the first media field; these
+/// "session-level" attributes convey additional information that applies
+/// to the conference as a whole rather than to individual media; an
+/// example might be the conference's floor control policy.
+///
+/// Attribute fields may be of two forms:
+//
+//   o property attributes.  A property attribute is simply of the form
+//     "a=<flag>".  These are binary attributes, and the presence of the
+//     attribute conveys that the attribute is a property of the session.
+//     An example might be "a=recvonly".
+//
+//   o value attributes.  A value attribute is of the form
+//     "a=<attribute>:<value>".  An example might be that a whiteboard
+//     could have the value attribute "a=orient:landscape"
+//
+//   Attribute interpretation depends on the media tool being invoked.
+//   Thus receivers of session descriptions should be configurable in
+//   their interpretation of announcements in general and of attributes in
+//   particular.
+//
+//   Attribute names must be in the US-ASCII subset of ISO-10646/UTF-8.
+/// v
+abstract class Attribute extends Field {
+
+  String _attribute;
+
+  String _value;
+
+  final int _type = 'a'.codeUnitAt(0);
```

Comment:
What's the reasoning behind this decision of defining `_type` as an int?

Reply:
In original implementation (Java) is Char, but there's no Char type. A integer type is often used as "char" because 'a' and "a" are Strings; that's as close as I could get to a Char. I chose to write 'a'.code...because it's more human readable instead of decimal 97. Could you suggest some alternative?

Reply:
Ahh, I totally understand your reasoning now.
Maybe we can use an enumeration type for this, but that seems overkill.

I think the best approach might be simply working with constants for each possible type value, e.g. `const a = 97` then using this constant as much as possible, converting to string whenever necessary.

Reply:
Following my advice above, not much would change. Basically we'd just replace all ocurrences of `'a'.codeUnitAt(0)` by usages of this newly defined `a` constant.

I think this is a small change that improves things quite a lot by removing a potential "eyebrow-raising" question on the redundancy of calling a string operation `codeUnitAt` passing zero as argument on a zero-length string for the sake of readability.

Using a constant is simpler, achieves the readability we want and avoids unnecessary questions :)

Reply:
Tks. I really like 'const a = 97' approach. I would use enum, but the way 'APPLE(“red”)' (enum with values) is not available in Dart yet.

Indeed. Writing like `const a = 97` is better. =D